app/Providers/MediawikiServiceProvider.php

```diff
@@ -46,7 +47,10 @@ public function register()
         });
 
         $this->app->bind(UserCreator::class, function ($app) {
-            return $app->make(MediawikiFactory::class)->newUserCreator();
+            $mw = $app->make(MediawikiFactory::class);
```

Comment:
Happy for this change to go in, though seems unrelated to this particular issue.